### ISSUE TITLE
image result for demosaic.cpp

### ISSUE THREAD

**Issue by Macdeini** [karansher/computer-graphics-raster-images#30]
Hi,

If the example image in the README.md for src/demosaic.cpp accurate for the final result? My image result looks this 
![image](https://user-images.githubusercontent.com/84424393/190507045-cff2b2c8-7ae1-499a-baff-e469419e3d1c.png)

and mine  more closely looks like the image examples on wikipedia
![image](https://user-images.githubusercontent.com/84424393/190507181-af67c449-8a7e-4e68-aca1-eaefe9e2af8a.png)
https://en.wikipedia.org/wiki/Bayer_filter

thanks

**Comment by abhimadan** [comment]
I suppose it looks like image 3 in that example, but that's just the Bayer mosaic values copied to their corresponding colour channels without reconstructing the other two channels. You need it to look like image 4, or in other words it should look almost identical to the original image.

**Comment by Macdeini** [comment]
I guess I'm missing something, could you explain the demosaicing algorithm? My understanding is that the missing rgb values for a pixel will be found by the immediable surronding pixels. So for example on a green pixel we would take the green value already present and the red and blue values are the average of the adjacent red and blue pixels, thats what I tried to do for the image i sent

**Comment by abhimadan** [comment]
Yes, that's the correct idea. It looks like you have a bug though, since the image is dominated by green, which, based on the image you posted, is what you get without looking at neighbours for the two missing colour channels.
